crates/deltalake-core/src/delta_datafusion/cdf/scan.rs

```diff
@@ -0,0 +1,318 @@
+    |id |name  |birthday  |_change_type    |_commit_version|_commit_timestamp      |
+    +---+------+----------+----------------+---------------+-----------------------+
+    |7  |Dennis|2023-12-29|update_postimage|2              |2023-12-29 16:41:33.843|
+    |5  |Emily |2023-12-29|update_postimage|2              |2023-12-29 16:41:33.843|
+    |7  |Dennis|2023-12-24|update_preimage |2              |2023-12-29 16:41:33.843|
+    |5  |Emily |2023-12-24|update_preimage |2              |2023-12-29 16:41:33.843|
+    |6  |Carl  |2023-12-29|update_postimage|2              |2023-12-29 16:41:33.843|
+    |3  |Dave  |2023-12-22|update_postimage|1              |2023-12-22 12:10:21.688|
+    |4  |Kate  |2023-12-22|update_postimage|1              |2023-12-22 12:10:21.688|
+    |6  |Carl  |2023-12-24|update_preimage |2              |2023-12-29 16:41:33.843|
+    |2  |Bob   |2023-12-22|update_postimage|1              |2023-12-22 12:10:21.688|
+    |3  |Dave  |2023-12-23|update_preimage |1              |2023-12-22 12:10:21.688|
+    |4  |Kate  |2023-12-23|update_preimage |1              |2023-12-22 12:10:21.688|
+    |2  |Bob   |2023-12-23|update_preimage |1              |2023-12-22 12:10:21.688|
+    |7  |Dennis|2023-12-29|delete          |3              |2024-01-06 11:44:59.624|
+    |8  |Claire|2023-12-25|insert          |0              |2023-12-22 12:10:18.922|
+    |7  |Dennis|2023-12-24|insert          |0              |2023-12-22 12:10:18.922|
+    |1  |Steve |2023-12-22|insert          |0              |2023-12-22 12:10:18.922|
+    |5  |Emily |2023-12-24|insert          |0              |2023-12-22 12:10:18.922|
+    |3  |Dave  |2023-12-23|insert          |0              |2023-12-22 12:10:18.922|
+    |4  |Kate  |2023-12-23|insert          |0              |2023-12-22 12:10:18.922|
+    |10 |Borb  |2023-12-25|insert          |0              |2023-12-22 12:10:18.922|
+    |6  |Carl  |2023-12-24|insert          |0              |2023-12-22 12:10:18.922|
+    |2  |Bob   |2023-12-23|insert          |0              |2023-12-22 12:10:18.922|
+    |9  |Ada   |2023-12-25|insert          |0              |2023-12-22 12:10:18.922|
+    +---+------+----------+----------------+---------------+-----------------------+
+    **/
+
+    #[tokio::test]
+    async fn test_load_local() -> TestResult {
+        // I checked in a pre-built table from spark, once the writing side is finished I will circle back to make these
+        // tests self contained and not rely on a physical table with them
+        let _table = DeltaOps::try_from_uri("./tests/data/cdf-table").await?;
+        let schema = _table.0.state.arrow_schema()?;
+        let partition_cols = _table.0.metadata()?.clone().partition_columns.clone();
+
+        let scan = DeltaCdfScan::new(_table.0.log_store.clone(), 0, schema, partition_cols);
+
+        let results = scan.scan().await?;
+        let data: Vec<RecordBatch> = collect_sendable_stream(results).await?;
+        print_batches(&data)?;
```

Comment:
Can you add an assert of the output of data?

Reply:
I'll flesh tests out to be more complete when we are more complete in the initial impl